src/app/AttributeAccessInterfaceRegistry.cpp

```diff
@@ -60,43 +58,50 @@ void UnregisterMatchingAttributeAccessInterfaces(F shouldUnregister)
 
 } // namespace
 
-void unregisterAttributeAccessOverride(AttributeAccessInterface * attrOverride)
+namespace chip {
+namespace app {
+
+AttributeAccessInterfaceRegistry & AttributeAccessInterfaceRegistry::Instance()
+{
+    static AttributeAccessInterfaceRegistry instance;
+    return instance;
+}
+
+void AttributeAccessInterfaceRegistry::Unregister(AttributeAccessInterface * attrOverride)
 {
-    gAttributeAccessInterfaceCache.Invalidate();
-    UnregisterMatchingAttributeAccessInterfaces([attrOverride](AttributeAccessInterface * entry) { return entry == attrOverride; });
+    mAttributeAccessInterfaceCache.Invalidate();
+    UnregisterMatchingAttributeAccessInterfaces([attrOverride](AttributeAccessInterface * entry) { return entry == attrOverride; },
+                                                mAttributeAccessOverrides);
 }
 
-void unregisterAllAttributeAccessOverridesForEndpoint(EmberAfDefinedEndpoint * definedEndpoint)
+void AttributeAccessInterfaceRegistry::UnregisterAllForEndpoint(EndpointId endpointId)
 {
     UnregisterMatchingAttributeAccessInterfaces(
```

Comment:
OK, I see.  This just never invalidated the cache.  But it needs to, no?

Followup is OK for this, since it's pre-existing.

Reply:
Added it to this PR ... may as well fix things.
I think there was a chance we never noticed because we unregister only on EP disabling, at which point the AAI is not usable anyway (and future enabling re-register things that invalidate the cache).